airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/task/DestinationTaskLauncher.kt

```diff
@@ -217,39 +200,13 @@ class DestinationTaskLauncher(
             launch(openStreamTaskFactory.make())
         }
 
-        if (loadPipeline != null) {
-            log.info { "Setting up load pipeline" }
-            loadPipeline.start { launch(it) }
-            log.info { "Launching update batch task" }
-            val updateBatchTask = updateBatchTaskFactory.make(this)
-            launch(updateBatchTask)
-            log.info { "Launching heartbeat task" }
-            launch(heartbeatTask)
-        } else {
-            if (!legacyFileTransfer) {
-                // Start a spill-to-disk task for each record stream
-                catalog.streams.forEach { stream ->
-                    log.info { "Starting spill-to-disk task for $stream" }
-                    val spillTask = spillToDiskTaskFactory.make(this, stream.descriptor)
-                    launch(spillTask)
-                }
-
-                repeat(config.numProcessRecordsWorkers) {
-                    log.info { "Launching process records task $it" }
-                    val task = processRecordsTaskFactory.make(this)
-                    launch(task)
-                }
-
-                repeat(config.numProcessBatchWorkers) {
-                    log.info { "Launching process batch task $it" }
-                    val task = processBatchTaskFactory.make(this)
-                    launch(task)
-                }
-            } // else legacy file transfer will be started by the load strategy
-            // Start flush task
-            log.info { "Starting timed file aggregate flush task " }
-            launch(flushTickTask)
-        }
+        log.info { "Setting up load pipeline" }
+        loadPipeline!!.start { launch(it) }
```

Comment:
nit could probably do something like this, to avoid the `!!`
```kotlin
class DTL(loadPipeline: LoadPipeline?) {
  private val loadPipeline: LoadPipeline
  init { loadPipeline = loadPipeline ?: throw }
}
```